File: production-cms-backend-complete/backend/src/entities/Content.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, ManyToOne, JoinColumn } from "typeorm";
import { MinLength, MaxLength, IsString, IsBoolean } from "class-validator";
import { User } from "./User";
import { Category } from "./Category";

export enum ContentStatus {
    DRAFT = "DRAFT",
    PUBLISHED = "PUBLISHED",
    ARCHIVED = "ARCHIVED"
}

@Entity("contents")
export class Content {
    @PrimaryGeneratedColumn("uuid")
    id!: string;

    @Column({ unique: true })
    @MinLength(5, { message: "Title must be at least 5 characters long" })
    @MaxLength(255, { message: "Title cannot exceed 255 characters" })
    title!: string;

    @Column({ type: "text" })
    @MinLength(20, { message: "Content body must be at least 20 characters long" })
    body!: string;

    @Column({ nullable: true })
    thumbnailUrl?: string;

    @Column({ type: "enum", enum: ContentStatus, default: ContentStatus.DRAFT })
    status!: ContentStatus;

    @Column({ default: false })
    isFeatured!: boolean;

    @ManyToOne(() => User, user => user.contents, { onDelete: "SET NULL" })
    @JoinColumn({ name: "authorId" })
    author!: User;

    @Column({ type: "uuid", nullable: true }) // Explicitly define column for authorId
    authorId!: string;

    @ManyToOne(() => Category, category => category.contents, { onDelete: "SET NULL" })
    @JoinColumn({ name: "categoryId" })
    category!: Category;

    @Column({ type: "uuid", nullable: true }) // Explicitly define column for categoryId
    categoryId!: string;

    @CreateDateColumn({ type: "timestamp", default: () => "CURRENT_TIMESTAMP" })
    createdAt!: Date;

    @UpdateDateColumn({ type: "timestamp", default: () => "CURRENT_TIMESTAMP", onUpdate: "CURRENT_TIMESTAMP" })
    updatedAt!: Date;
}

File: production-cms-backend-enhanced/src/main/java/com/cms/system/model/Content.java
package com.cms.system.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "content")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Content {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String body;

    private String slug; // URL-friendly identifier

    @Enumerated(EnumType.STRING)
    private ContentStatus status = ContentStatus.DRAFT;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    private User author;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum ContentStatus {
        DRAFT, PUBLISHED, ARCHIVED
    }

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
        if (this.slug == null || this.slug.isEmpty()) {
            this.slug = generateSlug(this.title);
        }
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
        if (this.slug == null || this.slug.isEmpty()) {
            this.slug = generateSlug(this.title);
        }
    }

    private String generateSlug(String title) {
        return title.toLowerCase().replaceAll("[^a-z0-9\\s-]", "").replaceAll("[\\s-]+", " ").trim().replaceAll(" ", "-");
    }
}